src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java

```diff
@@ -292,14 +292,11 @@ private static boolean isInNoBraceControlStatement(DetailAST ast) {
      * @return whether the context of the assignment AST indicates the idiom
      */
     private static boolean isInLoopIdiom(DetailAST ast) {
-        boolean result = false;
```

Comment:
After changing `if` to `&&` one uncovered branch was reported by Jacoco. New test case added.